ligo/swaps.mligo

```diff
@@ -165,3 +165,33 @@ let y_to_x (s : storage) (p : y_to_x_param) : result =
         let op_receive_y = y_transfer Tezos.sender Tezos.self_address dy_spent in
         let op_send_x = x_transfer Tezos.self_address p.to_dx dx_received in
         ([op_receive_y ; op_send_x], s_new)
+
+
+(* Trade X with X' in a different contract. *)
+let x_to_x_prime (s : storage) (p : x_to_x_prime_param) : result =
+
+    (* Ensure provided contract address is correct. *)
+    let x_prime_contract =
+        match (Tezos.get_entrypoint_opt "%y_to_x" p.x_prime_contract
+            : y_to_x_param contract option) with
+        | Some contract -> contract
+        | None -> (failwith("TODO: incorrect segmented cfmm contract") : y_to_x_param contract) in
```

Comment:
There's no way to avoid this check, so we should indeed add an error code for this scenario

Reply:
I've added the error.